### PR TITLE
Assault borgs now resist flashes and flashbangs

### PR DESCRIPTION
**Pull request by Mqiib**
Exactly what it says on the tin: assault cyborgs are now immune to standard flashes. No more baby bitch dying to a random engiborg with a flash that snuck up on it! These motherfuckers will KICK YOUR ASS for disrespecting them with your silly little lightbulb.
This does not make them immune to EMPs, but DOES also make them immune to flashbangs. They're way more expensive than the other two borgs, they deserve a buff.

# Wiki Documentation

Probably important to mention that assault borgs can't be flashed.

# Changelog

:cl:  
tweak: Assault cyborgs are no longer stunned by flashes or flashbangs.
/:cl:

### CODE REVIEW

**Comment by tattax** [comment]
less stun time on everything instead of just ignoring flashes and flashbangs would be a better change, someone can just rush the ion gun and die first then what you have is the hos gun that has three shots of ion. syndiborgs have grenade launchers that can force people out of corners to preventing flashing and stuff, flashbang is less of an issue since you should stay with your team or someone else anyway!

**Comment by tattax** [comment]
also grenade launchers activate flash bangs ^^

**Comment by N3D6** [comment]
Powercreep

**Comment by N3D6** [comment]
Wait never mind I kind of actually like this idea

**Comment by maxion12345** [comment]
assault borgs are expensive so this makes sense
always were SUPER Cheeseable with the flashbangs

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -1356,3 +1363,6 @@
 	var/datum/computer_file/program/robotact/program = modularInterface.get_robotact()
 	if(program)
 		program.force_full_update()
+
+/mob/living/silicon/robot/get_eye_protection()
```
Comment (ToasterBiome) [inline code/modules/mob/living/silicon/robot/robot.dm:1367]: is this actually called anywhere?

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by Runian** [comment]
wow. buffing borgs. we're suppose to hate borgs. its the borg's fault for going near people who have flashes when they have a ranged gun to spam. flashbangs are meh because you stun your crew mates (who don't have protection) just so you can stun the borg.

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (Mqiib) [reply code/modules/mob/living/silicon/robot/robot.dm:1367]: I tested it and it works. It's in flash_act, normally just returns 0 for silicons

**Comment by Bibby0110** [comment]
Just support the borg so it doesn't end up alone with 3 sec officers and a flashbang?

**Comment by Marmio64** [comment]
I'd say assault Borgs should resist the stuns but not the blindness

**Review by ToasterBiome — APPROVED** [review]
code fine, up to jamie

**Comment by bruhlookatthisdood** [comment]
maybe the flashbang and flashes should have the same effect that the laser pointer does where it turns their screen into static and temporarily removes their modules?

**Comment by JamieD1** [comment]
> maybe the flashbang and flashes should have the same effect that the laser pointer does where it turns their screen into static and temporarily removes their modules?

Yeah I like this idea, Do this instead. Make it so the borg has to retreat before reengaging. Prevents a hard stun (Death). But means the borg cant just keep being a twit to people.

**Review by ToasterBiome — DISMISSED** [review]
do as the jamie asks

**Comment by JamieD1** [comment]
Hows this going?

**Comment by Moja10** [comment]
Also I think nukies can get flashbangs somehow, I think some spawn on their base. I have seen nukie borgs die to bald nukies who use these flashbangs. Justice for assault borgs!!!

**Comment by JamieD1** [comment]
Hows this going? Check 2.

**Comment by JamieD1** [comment]
Hows this Going? Check 3.

**Comment by Mqiib** [comment]
> Hows this Going? Check 3.

It is done. Flashes disarm them for a full 5 seconds and flashbangs for 3 (because they only stunned for 3 anyways).
Also I did actually test it and it does work.

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**